docs/SPECS.md

```diff
@@ -0,0 +1,1435 @@
+in Golang. Servers running drand can be linked with each other to produce
+collective, publicly verifiable, unbiased, unpredictable random values at fixed
+intervals using bilinear pairings and threshold cryptography. Drand nodes can
+also serve locally-generated private randomness to clients.
+
+This document is a specification of the drand protocols.
+
+## Table of Contents
+
+* [Notations](#notations)
+   * [Drand node](#drand-node)
+   * [Drand beacon](#drand-beacon)
+   * [Group configuration](#group-configuration)
+      * [Group Configuration Hash](#group-configuration-hash)
+* [Wireformat &amp; API](#wireformat--api)
+* [Drand Modules](#drand-modules)
+   * [Setup phase](#setup-phase)
+      * [Collecting the keys of the participants](#collecting-the-keys-of-the-participants)
+      * [Coordinator pushing the new group configuration](#coordinator-pushing-the-new-group-configuration)
+   * [Distributed Key Generation](#distributed-key-generation)
+      * [Types of distributed key generation](#types-of-distributed-key-generation)
+      * [Network level packets](#network-level-packets)
+      * [Phase transitions](#phase-transitions)
+      * [Deal Phase](#deal-phase)
+      * [Response Phase](#response-phase)
+      * [Justification Phase](#justification-phase)
+      * [Finish Phase](#finish-phase)
+   * [Randomness generation](#randomness-generation)
+      * [Overiew](#overiew)
+      * [Randomness Generation Period](#randomness-generation-period)
+      * [Beacon Chain](#beacon-chain)
+      * [Catchup mode](#catchup-mode)
+      * [Syncing](#syncing)
+* [Cryptographic specification](#cryptographic-specification)
+   * [Drand Curve](#drand-curve)
+   * [Distributed Public Key](#distributed-public-key)
+   * [Beacon Signature](#beacon-signature)
+   * [Partial Beacon Signature](#partial-beacon-signature)
+   * [Distributed Key Generation](#distributed-key-generation-1)
+      * [Input](#input)
+      * [Authentication](#authentication)
+      * [Setup](#setup)
+      * [Deal Phase](#deal-phase-1)
+      * [Response Phase](#response-phase-1)
+      * [Justification Phase](#justification-phase-1)
+      * [Finish Phase](#finish-phase-1)
+   * [Resharing](#resharing)
+      * [Setup](#setup-1)
+      * [Distinction of Roles](#distinction-of-roles)
+      * [Deal Phase](#deal-phase-2)
+      * [Response Phase](#response-phase-2)
+      * [Justification Phase](#justification-phase-2)
+      * [Finish Phase](#finish-phase-2)
+* [External API](#external-api)
+* [Control API](#control-api)
+* [Appendix A. DKG packets](#appendix-a-dkg-packets)
+* [THINGS TO REVIEW](#things-to-review)
+
+## Notations
+
+### Drand node
+
+A drand node is a server that runs the drand code, that participates in the
+distributed key generations phases, in the randomness generation and that can
+reply to public request API. A Go struct representation is as follow:
+```go
+type Node struct {
```

Comment:
this is an odd representation because it's clearly incomplete - no private key - but has seemingly not-relevant-to-protocol details like `TLS bool`. What is it supposed to be demonstrating?

Reply:
It's the public side of a node - i've added details.